### PR TITLE
feat: add mkcalendar & Report to body methods in next

### PR DESCRIPTION
**Pull request by keyservlad**
#### Checklist

For context see #5439 

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/httpMethods.js
Comment (gurgunday) [inline lib/httpMethods.js:30]: This is normally in alphabetical order

Also just realized 'MKCOL' should be above 'MOVE' for the same reason

**Comment by gurgunday** [comment]
The rest of the changes we can pull from main I think, but maybe we should add their shorthands here as well

Like this: https://github.com/fastify/fastify/pull/4409

**Comment by keyservlad** [comment]
I merged main and added the route shorthand methods. 
Still getting the route-shorthand test failing tho ahah ;D

**Comment by gurgunday** [comment]
yeah looks great, just one issue, because we use squash merge, I think we should merge main to next first, and merge this after that, otherwise all these merge changes will end up being conflicts

**Review by gurgunday — APPROVED** [review]
lgtm

**Comment by keyservlad** [comment]
@gurgunday Thank you guys very much for your work. I noticed you didn't include the route shorthand changes for the new methods. If you ever want to implement them, you can find the changes needed in this PR.

Here are the files to change :
docs/Reference/Routes.md

test/types/route.test-d.ts
instance.d.ts

**Comment by gurgunday** [comment]
Good spot, the shorthands were added, but you are right that the types weren't, opening the PR again to not forget

**Comment by keyservlad** [comment]
I will try to rebase and get rid of the commits that come from the merge if that's ok with you

**Comment by gurgunday** [comment]
Yeah but I’d say let’s wait for the merge to main of v5 first, like @jsumners said, or maybe he’ll add it manually

**Comment by gurgunday** [comment]
Again, sorry for the trouble, can you open the type PR again? It can be a new PR to make things easier, and it should target main now

**Pull request closed without merging**
